File: backend/ecommerce/src/main/java/com/code/ecommerce/service/search/SearchService.java
package com.code.ecommerce.service.search;

import com.code.ecommerce.dto.requests.itemSearch.ItemSearchRequest;
import com.code.ecommerce.dto.requests.orderSearch.OrderSearchRequest;
import com.code.ecommerce.dto.response.ItemResponse;
import com.code.ecommerce.dto.response.itemSearch.ItemCatalogueResponse;
import com.code.ecommerce.dto.response.orderSearch.OrderSearchResponse;
import org.springframework.data.domain.Page;

public interface SearchService {
    Page<ItemResponse> searchItems(ItemSearchRequest request);  // searching the items based on the request parameters
    Page<OrderSearchResponse> searchOrder(OrderSearchRequest request);   // searching the order based on the request parameters
}

File: frontend/ecommerce/src/common/api/searchService.ts
import { ItemSearchRequest, OrderSearchRequest } from "../types/search";
import api from "./axiosconfig";
import { API_ROUTES } from "./apiRoutes";

export const searchItems = async(request: ItemSearchRequest) => {
    const response = await api.post(API_ROUTES.SEARCH.ITEM_SEARCH, request);  // calling the api for item search 
    if(response)
        return response.data.data;  // returning the data of the item search filter
}

export const searchOrders = async(request: OrderSearchRequest) => {
    const response = await api.post(API_ROUTES.SEARCH.ORDER_SEARCH, request);  // calling the api for order search
    if(response)
        return response.data.data;  // returning the data for the order search filter
}